src/keywords.rs

```diff
@@ -973,3 +973,61 @@ pub const RESERVED_FOR_IDENTIFIER: &[Keyword] = &[
     Keyword::STRUCT,
     Keyword::TRIM,
 ];
+
+pub const NA: usize = usize::MAX;
+
+#[rustfmt::skip]
+pub const KEYWORD_LOOKUP_INDEX_ROOT: &[usize; 26] = &[
```

Comment:
I think this is not super maintainable (needs to be updated manually when adding keywords).

Reply:
I agree - it we go with this type of table driven approach we should have some sort of update script (or build.rs) that builds the table from the enum.